src/neptune/metadata_containers/metadata_container.py

```diff
@@ -175,14 +178,17 @@ def __init__(
             env_name=NEPTUNE_ENABLE_DEFAULT_ASYNC_NO_PROGRESS_CALLBACK,
         )
 
-        self._op_processor: OperationProcessor = get_operation_processor(
-            mode=mode,
-            container_id=self._id,
-            container_type=self.container_type,
-            backend=self._backend,
-            lock=self._lock,
-            flush_period=flush_period,
-            queue=self._signals_queue,
+        self._op_processor: LazyOperationProcessorWrapper = LazyOperationProcessorWrapper(
```

Comment:
> I guess we could define self._op_processor of type OperationProcessor and in __init__ set it as non wrapped processor and in child fork set it as LazyWrapper so we just switch implementation of interface. What do you think about that?

Yes, as Lazy should be just a special implementation of `OperationProcessor` interface/abstract class this won't hurt as as long as the typing was set to this interface.

> there is no real difference (init will be triggered by self._write_initial_attributes() few lines later ) if I wrap op_processor in LazyWrapper here.

There is one difference called ... debugging 🤖 , or rather the complexity of it. In case of having to follow some quite low-level things with this approach, you will have to go a way deeper in a calls stack etc.